lib/response_massager.js

```diff
@@ -61,10 +61,14 @@ class ResponseMassager {
 
   massagedResponse (request, options = {}) {
     let response = this.elasticSearchResponse
-    const allItemsBibQuery = request?.query?.all_items
+    const queryKeys = Object.keys(request?.query || {})
+    const itemFilterQuery = queryKeys
+      .filter((param) => param.startsWith('item_')).length > 0
+    const returnAllItems = queryKeys?.includes('all_items') &&
+      !itemFilterQuery
```

Comment:
This is fine, but I kind of think use of `?all_items` with any item filter should return a 4xx earlier, since it's nonsensical. That would mean we don't have to handle that possibility here. As written, it looks like we'll just ignore `all_items` if any item filter is applied. But above is also fine.

Reply:
Ah yes there was a decision made around this that I forgot to explain. The front end spec is that a user can filter items, and then click view all to de-paginate the results. We decided to expedite the release of this endpoint by letting the front end decide whether to use all_items or to revert to the batched filtering situation. Eventually, we actually do want to have some combination of params that will return all of a bib's items for a given filter.